### PR TITLE
add key control for TEST variable(params) when is not set, control the exception

### PR DESCRIPTION
**Pull request by MalloZup**
[script/client]

### CODE REVIEW

File: script/client [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
```
Comment (okurz) [inline script/client]: that should not be part of your change

File: script/client [hunk 2]
```diff
@@ -111,6 +111,16 @@ if ($options{params}) {
...
+    ## FIXME:
+    ## if we don't set the TEST variable we get:
+    ## "org.freedesktop.DBus.Error.Failed: DBIx::Class::Storage::DBI::_dbh_execute(): DBI Exception: DBD::SQLite::st execute failed: NOT NULL constraint failed:
+    ## curiosly i tested with /usr/share/openqa/script/client --params wrongTEST.json jobs post
+    ## wrongTEST look like  "DISTRI" : "sle", "VERSION" : "Migration", "BUILD" : "GM", "ARCH" : "x86_64", "HDD_1" : "sles.iso"
+    ## and only the missing TEST variable is causing this exception. other variables don't cause the Exception, and user can
+    ## this is fix already the problem but , it will be better fo fix maybe on the right place. (?)
+    if (!exists($params{TEST})) {
+        die "variable \"TEST\:\"NAME__TEST\":variable must be set on *.json file\n";
+    }
```
Comment (okurz) [inline script/client]: IMHO your intentions are good but checks for the existance of variables should be where they are used and therefore more closely connected.

**Comment by MalloZup** [comment]
yes, i know, that , and i added this already in my comment. For the moment i know a few internals thing of the arch of Openqa. i think @ usr/share/openqa/lib/OpenQA/WebAPI/Controller is the right place, but anyway i think that also in the script part is a good place to control that.

**Comment by coolo** [comment]
This idles now for a month - and I must say I don't like it at all. The validation should happen in the server and the client should report errors correctly. This is just a wild hack for a very specific error that happened - there are way more errors that can happen and the server needs to validate anyway.

**Pull request closed without merging**